src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts

```diff
@@ -221,31 +222,36 @@ function createMirrorNodeService() {
       }
     );
     */
+    const params: any = {
+      order: "desc",
+      /**
+       * Every proposal fetched from the mirror node logs requires an additional hashio
+       * JSON RPC call to fetch the remaining details for the UI. Some of these calls are failing
+       * with a "HBAR rate limit exceeded" error resulting in the UI displaying an error. This is due to
+       * our queries exceeding the global rate limit for the JSON RPC service.
+       *
+       * To reduce the frequency at which this error occurs a temporary limit has been put in place. A
+       * max of 5 proposal details for each type of proposal will be fetched from the proposal logs.
+       *
+       * @see {@link file://./../../../../architecture/05_Event_Based_Historical_Queries.md} for more details
+       * regarding a long term solution.
+       */
+      limit: 5,
+    };
+    if (!limitResults) {
+      delete params.limit;
+    }
     const response = await testnetMirrorNodeAPI.get(`/api/v1/contracts/${contractId.toString()}/results/logs`, {
-      params: {
-        order: "desc",
-        /**
-         * Every proposal fetched from the mirror node logs requires an additional hashio
-         * JSON RPC call to fetch the remaining details for the UI. Some of these calls are failing
-         * with a "HBAR rate limit exceeded" error resulting in the UI displaying an error. This is due to
-         * our queries exceeding the global rate limit for the JSON RPC service.
-         *
-         * To reduce the frequency at which this error occurs a temporary limit has been put in place. A
-         * max of 5 proposal details for each type of proposal will be fetched from the proposal logs.
-         *
-         * @see {@link file://./../../../../architecture/05_Event_Based_Historical_Queries.md} for more details
-         * regarding a long term solution.
-         */
-        limit: 5,
-      },
+      params,
     });
```

Comment:
We already had fetchEvents `function` in place any special case we needed this one.?